components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -521,6 +522,10 @@ class GeckoEngineSession(
         }
 
         override fun onFocusRequest(session: GeckoSession) = Unit
+
+        override fun onWebAppManifest(session: GeckoSession, manifest: JSONObject) {
+            notifyObservers { onWebAppManifestLoaded(manifest) }
```

Comment:
Let's consider moving `WebAppManifest` to `concept-engine`. Then we can pass the high-level object along early. What do you think?